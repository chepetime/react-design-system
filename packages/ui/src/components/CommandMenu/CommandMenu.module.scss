@use "./../../styles" as styles;

:root {
  --commandmenu-background-color: #{styles.$pink-500};
}

:global .dark {
  --commandmenu-background-color: #{styles.$indigo-500};
}

.CommandMenu {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--commandmenu-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
