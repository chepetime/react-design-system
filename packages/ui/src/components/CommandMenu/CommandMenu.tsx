import React from "react";
import classnames from "classnames";
import css from "./CommandMenu.module.scss";

export interface CommandMenuProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const CommandMenu = React.forwardRef<HTMLDivElement, CommandMenuProps>(
  (props: CommandMenuProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.CommandMenu)} ref={ref}>
        <p>CommandMenu component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

CommandMenu.displayName = "CommandMenu";
