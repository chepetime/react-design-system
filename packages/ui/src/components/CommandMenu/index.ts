export { CommandMenu } from "./CommandMenu";
export type { CommandMenuProps } from "./CommandMenu";