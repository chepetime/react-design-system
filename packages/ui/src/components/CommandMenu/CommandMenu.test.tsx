
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { CommandMenu } from "./CommandMenu";
import { render, screen } from "@testing-library/react";

describe("CommandMenu", () => {
  it("renders without errors", () => {
    render(<CommandMenu />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<CommandMenu>Some children</CommandMenu>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<CommandMenu>Some children</CommandMenu>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<CommandMenu />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
