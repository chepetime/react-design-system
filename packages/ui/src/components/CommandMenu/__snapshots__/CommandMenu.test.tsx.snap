// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandMenu matches snapshot 1`] = `
<div>
  <div
    class="CommandMenu"
  >
    <p>
      CommandMenu component working!
    </p>
    <div />
  </div>
</div>
`;
